broker/cmd/dejaqcli-consumer/main.go

```diff
@@ -0,0 +1,171 @@
+		return errors.New("topic, overseerSeed and consumer ID are mandatory values")
+	}
+	if c.MaxBufferSize < 1 {
+		return errors.New("MaxBufferSize has to be > 0")
+	}
+	if _, err := time.ParseDuration(c.LeaseDuration); err != nil {
+		return fmt.Errorf("LeaseDuration provided but wrong value %s", err.Error())
+	}
+	if _, err := time.ParseDuration(c.UpdatePreloadStatsTick); err != nil {
+		return fmt.Errorf("UpdatePreloadStatsTick provided but wrong value %s", err.Error())
+	}
+	if _, err := time.ParseDuration(c.TimeoutDuration); err != nil {
+		return fmt.Errorf("timeout provided but wrong value %s", err.Error())
+	}
+	return nil
+}
+
+func (c *Config) durationLease() time.Duration {
+	r, _ := time.ParseDuration(c.LeaseDuration)
+	return r
+}
+
+func (c *Config) durationPreloadStatsTick() time.Duration {
+	r, _ := time.ParseDuration(c.UpdatePreloadStatsTick)
+	return r
+}
+
+func (c *Config) durationTimeout() time.Duration {
+	r, _ := time.ParseDuration(c.TimeoutDuration)
+	return r
+}
+
+func main() {
+	logger := logrus.New().WithField("component", "consumer")
+
+	c := &Config{}
+	err := cleanenv.ReadEnv(c)
+	if err != nil {
+		logger.Fatal(err)
+	}
+	if err = c.IsValid(); err != nil {
+		logger.Fatal(err)
+	}
+
+	ctx, shutdownEverything := context.WithCancel(context.Background())
+	if c.TimeoutDuration != "" {
+		ctx, _ = context.WithTimeout(ctx, c.durationTimeout())
+	}
+
+	clientConfig := satellite.Config{
+		Cluster:        "",
+		OverseersSeeds: []string{c.OverseerSeed},
+	}
+	client, err := satellite.NewClient(ctx, logger, &clientConfig)
+	if err != nil {
+		logger.WithError(err).Fatal("brokerClient")
+	}
+
+	waitForCtx := ctx
+	if c.TimeoutDuration == "" {
+		waitForCtx, _ = context.WithTimeout(ctx, time.Second*10)
+	}
+	if !client.WaitForConnection(waitForCtx) {
+		logger.Fatal("The connection to the broker cannot be established in time.")
+	}
+
+	cons := client.NewConsumer(&consumer.Config{
+		ConsumerID:             c.ConsumerID,
+		Topic:                  c.Topic,
+		Cluster:                "",
+		MaxBufferSize:          int64(c.MaxBufferSize),
+		LeaseDuration:          c.durationLease(),
+		UpdatePreloadStatsTick: c.durationPreloadStatsTick(),
+	})
+	err = cons.Handshake(ctx)
+	if err != nil {
+		logger.WithError(err).Fatal("cannot handshake")
+	}
+
+	c.strategy = sync_consume.StrategyContinuous
+	if c.StopAfterCount > 0 {
+		c.strategy = sync_consume.StrategyStopAfter
+	}
+	cc := sync_consume.SyncConsumeConfig{
+		Strategy:        c.strategy,
+		StopAfterCount:  c.StopAfterCount,
+		DeleteMessages:  true,
+		DecreaseCounter: nil,
+	}
+
+	go func() {
+		result, err := sync_consume.Consume(ctx, logger, cons, &cc)
+		if err != nil {
+			// TODO check wat returns here
+			switch v := err.(type) {
+			case derror.MessageIDTupleList:
+				for _, ve := range v {
+					logger.WithError(ve.MsgError).Errorf("message ID failed %s", string(ve.MsgID))
+				}
+			default:
+				logger.Error(err.Error())
+			}
+			return
```

Comment:
this should not exit here, it needs to do the shutdown or move the shutdown in a defer

Reply:
I choose to use defer and do not log avg in error case (result could be nil)